### PR TITLE
refactor: megagraph

### PR DESCRIPTION
**Pull request by antonyip**
Bounty: https://gitcoin.co/issue/Badger-Finance/badger-subgraph/11/100026321

Ethereum Subgraph: https://thegraph.com/legacy-explorer/subgraph/antonyip/badger_community_subgraph6?version=pending
Matic Subgraph: https://thegraph.com/legacy-explorer/subgraph/antonyip/badger_community_subgraph5?version=pending

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "badger-dao",
```
Comment (lookfirst) [inline package.json:2]: correct name?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "license": "MIT",
+  "scripts": {
+    "format": "echo 'Linter Breaks Code...'",
+    "format-orig": "yarn run lint --fix",
+    "lint": "eslint ."
+  },
+  "dependencies": {
```
Comment (lookfirst) [inline package.json:9]: some of these are deps and some are devDeps

File: subgraphs/badger-dao/src/constants.ts [hunk 3]
```diff
@@ -34,3 +34,10 @@ export enum SettType {
   v2,
   Affiliate,
 }
+
+// Order matters, same as in registry contract
+export let REGISTRY_SETT_STATUSES: string[] = [
```
Comment (lookfirst) [inline subgraphs/badger-dao/src/constants.ts:39]: const

File: subgraphs/badger-dao/src/constants.ts [hunk 4]
```diff
@@ -34,3 +34,10 @@ export enum SettType {
...
+  "experimental",
+  "guarded",
+  "open"
+]
```
Comment (lookfirst) [inline subgraphs/badger-dao/src/constants.ts:43]: missing ;

File: subgraphs/badger-dao/src/entities/affiliate-sett.ts [hunk 5]
```diff
@@ -26,6 +26,7 @@ export function loadAffiliateSett(address: Address): Sett {
     sett.netShareDeposit = ZERO;
     sett.grossShareDeposit = ZERO;
     sett.grossShareWithdraw = ZERO;
+    sett.status = "not applicable";
```
Comment (lookfirst) [inline subgraphs/badger-dao/src/entities/affiliate-sett.ts:29]: single quotes

File: subgraphs/badger-dao/src/entities/badger-sett-v2.ts [hunk 6]
```diff
@@ -26,6 +26,7 @@ export function loadSettV2(address: Address): Sett {
     sett.netShareDeposit = ZERO;
     sett.grossShareDeposit = ZERO;
     sett.grossShareWithdraw = ZERO;
+    sett.status = "dev";
```
Comment (lookfirst) [inline subgraphs/badger-dao/src/entities/badger-sett-v2.ts:29]: single quotes

File: subgraphs/badger-dao/package.json [hunk 7]
```diff
@@ -8,32 +8,23 @@
...
     "deploy:mainnet": "graph deploy --studio badger-dao",
     "deploy:matic": "graph deploy --node https://api.thegraph.com/deploy/ axejintao/badger-dao-polygon",
     "deploy:xdai": "graph deploy --node https://api.thegraph.com/deploy/ axejintao/badger-dao-xdai",
     "deploy:bsc": "graph deploy --node https://api.thegraph.com/deploy/ axejintao/badger-dao-bsc",
     "deploy:mainnet-staging": "graph deploy --node https://api.thegraph.com/deploy/ axejintao/badger-dao",
     "create-local": "graph create --node http://localhost:8020/ axejintao/badger-dao",
     "remove-local": "graph remove --node http://localhost:8020/ axejintao/badger-dao",
-    "deploy-local": "graph deploy --node http://localhost:8020/ --ipfs http://localhost:5001 axejintao/badger-dao",
-    "format": "yarn run lint --fix",
-    "lint": "eslint ."
```
Comment (lookfirst) [inline subgraphs/badger-dao/package.json:20]: why remove linting and formatting?

File: subgraphs/badger-dao/src/entities/badger-sett.ts [hunk 8]
```diff
@@ -35,3 +37,26 @@ export function loadSett(address: Address): Sett {
...
+}
+
+export function SettStatusString(status: i32): string {
+  let statusString = "unknown"
+  if (0 <= status && status < REGISTRY_SETT_STATUSES.length) {
+    statusString = REGISTRY_SETT_STATUSES[status]
+  }
+
+  return statusString as string
+}
```
Comment (lookfirst) [inline subgraphs/badger-dao/src/entities/badger-sett.ts:62]: code format

**Comment by lookfirst** [comment]
All of this needs the badger standard code formatting applied.

**Comment by axejintao** [comment]
thank you for your work, this was a great effort - the subgraph needed to be redone for some multichain implementations and so this was not accepted at it was done at this multi chain crossroads.

**Pull request closed without merging**